### PR TITLE
fqdn: test zombie liveness boundaries

### PR DESCRIPTION
**Pull request by henockt**
Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request]
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin]
- [x] Provide a title or release-note blurb suitable for the release notes.
- [x] Disclose use of machine learning models (including LLMs and other generative AI)
      in accordance with the [Cilium AI Policy], and indicate the rating using
      [AI Influence Level].
      Example: "This PR was prepared with AIL:3. I personally checked X."
- [x] Thanks for contributing!

Add tests for the grace-period deadline and the mark-alive boundary of isConnectionAlive ahead of the FQDN garbage collection refactor (#45849).

This PR was prepared with AIL:3. I reviewed and ran all the tests locally.

Ref: #45849

[AI Influence Level]: https://danielmiessler.com/blog/ai-influence-level-ail
[Cilium AI Policy]: https://github.com/cilium/community/blob/main/AI-POLICY.md
[Developer’s Certificate of Origin]: https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo
[Submitting a pull request]: https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request

### CODE REVIEW

**Comment by joestringer** [comment]
@cilium/fqdn could you take a look when you get a chance?

File: pkg/fqdn/cache_test.go [hunk 1]
```diff
@@ -1424,3 +1425,67 @@ func Test_sortZombieMappingSlice(t *testing.T) {
...
+
+		// One nanosecond past the grace period the zombie is dead.
+		zombies.SetCTGCTime(now.Add(grace+time.Nanosecond), now.Add(2*grace))
+		require.False(t, isAlive(zombies))
+		alive, dead := zombies.GC()
+		require.Empty(t, alive)
+		assertZombiesContain(t, dead, map[string][]string{"1.1.1.1": {"test.com"}})
+	})
+
+	t.Run("mark alive precedence", func(t *testing.T) {
```
Comment (vipul-21) [inline pkg/fqdn/cache_test.go]: Can we extend `TestZombiesDumpAlive` with the later unmarked CT GC step instead ? It already checks survival when `AliveAt == lastCTGCUpdate`; adding the subsequent expiry check there would consolidate the mark-alive sequence and keep this new test focused on the non-zero grace-period boundary

**Review by vipul-21 — CHANGES_REQUESTED** [review]
